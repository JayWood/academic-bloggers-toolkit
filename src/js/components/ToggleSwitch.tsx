import * as React from 'react';
import { createTooltip, destroyTooltip } from 'utils/Tooltips';

interface Props {
    checked: boolean;
    label: string;
    onChange(e: React.FormEvent<HTMLInputElement>): void;
}

export class ToggleSwitch extends React.PureComponent<Props, {}> {
    handleMouseOver = (e: React.MouseEvent<HTMLLabelElement>) => {
        createTooltip(e.currentTarget, e.currentTarget.getAttribute('data-tooltip')!, 'left');
    };

    render() {
        const { checked, onChange, label } = this.props;
        return (
            <div>
                <input
                    type="checkbox"
                    id="inline-toggle"
                    style={{ display: 'none' }}
                    checked={checked}
                    aria-checked={checked}
                    onChange={onChange}
                />
                <label
                    htmlFor="inline-toggle"
                    role="tooltip"
                    data-tooltip={label}
                    onMouseOver={this.handleMouseOver}
                    onMouseOut={destroyTooltip}
                />
                <style jsx>{`
                    label {
                        position: relative;
                        display: block;
                        margin: 0 10px;
                        height: 20px;
                        width: 44px;
                        background: #898989;
                        border-radius: 100px;
                        cursor: pointer;
                        -webkit-transition: all 0.3s ease;
                        transition: all 0.3s ease;
                    }
                    label:after {
                        position: absolute;
                        left: -2px;
                        top: -3px;
                        display: block;
                        width: 26px;
                        height: 26px;
                        border-radius: 100px;
                        background: #fff;
                        -webkit-box-shadow: 0px 3px 3px rgba(0, 0, 0, 0.05);
                        box-shadow: 0px 3px 3px rgba(0, 0, 0, 0.05);
                        content: '';
                        -webkit-transition: all 0.3s ease;
                        transition: all 0.3s ease;
                    }
                    label:active:after {
                        -webkit-transform: scale(1.15, 0.85);
                        transform: scale(1.15, 0.85);
                    }
                    input:checked ~ label {
                        background: #6f9eb1;
                    }
                    input:checked ~ label:after {
                        left: 20px;
                        background: #0085ba;
                    }
                    input:disabled ~ label {
                        background: #d5d5d5;
                        pointer-events: none;
                    }
                    input:disabled ~ label:after {
                        background: #bcbdbc;
                    }
                `}</style>
            </div>
        );
    }
}
