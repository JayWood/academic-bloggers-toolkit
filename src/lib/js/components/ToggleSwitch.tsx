import * as React from 'react';
import { createTooltip, destroyTooltip } from '../utils/Tooltips';

interface Props {
    checked: boolean;
    label: string;
    onChange(e: React.FormEvent<HTMLInputElement>): void;
}

export class ToggleSwitch extends React.PureComponent<Props, {}> {

    handleMouseOver = (e) => {
        createTooltip(e.target, e.target.getAttribute('data-tooltip'), 'left');
    }

    render() {
        const { checked, onChange, label } = this.props;
        return (
            <div>
                    <input
                        type="checkbox"
                        id="inline-toggle"
                        className="toggle"
                        style={{display: 'none'}}
                        checked={checked}
                        aria-checked={checked}
                        onChange={onChange}
                    />
                    <label
                        htmlFor="inline-toggle"
                        className="toggle-lbl"
                        role="tooltip"
                        data-tooltip={label}
                        onMouseOver={this.handleMouseOver}
                        onMouseOut={destroyTooltip}
                    />
                </div>
        );
    }
}
