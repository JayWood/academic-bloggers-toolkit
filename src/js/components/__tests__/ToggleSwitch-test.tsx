import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { ToggleSwitch } from '../ToggleSwitch';

const changeFn = () => void 0;

describe('<ToggleSwitch />', () => {
    it('should match snapshots', () => {
        let component = renderer.create(
            <ToggleSwitch checked={false} label="Test 1" onChange={changeFn} />
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        component = renderer.create(
            <ToggleSwitch checked={true} label="Test 2" onChange={changeFn} />
        );
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
