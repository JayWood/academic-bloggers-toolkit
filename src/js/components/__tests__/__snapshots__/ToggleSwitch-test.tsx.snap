// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleSwitch /> should match snapshots 1`] = `
<div>
  <input
    aria-checked={false}
    checked={false}
    id="inline-toggle"
    onChange={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    type="checkbox"
  />
  <label
    data-tooltip="Test 1"
    htmlFor="inline-toggle"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    role="tooltip"
  />
  <style
    jsx={true}
  >
    
                        label {
                            position: relative;
                            display: block;
                            margin: 0 10px;
                            height: 20px;
                            width: 44px;
                            background: #898989;
                            border-radius: 100px;
                            cursor: pointer;
                            -webkit-transition: all 0.3s ease;
                            transition: all 0.3s ease;
                        }
                        label:after {
                            position: absolute;
                            left: -2px;
                            top: -3px;
                            display: block;
                            width: 26px;
                            height: 26px;
                            border-radius: 100px;
                            background: #fff;
                            -webkit-box-shadow: 0px 3px 3px rgba(0, 0, 0, 0.05);
                            box-shadow: 0px 3px 3px rgba(0, 0, 0, 0.05);
                            content: '';
                            -webkit-transition: all 0.3s ease;
                            transition: all 0.3s ease;
                        }
                        label:active:after {
                            -webkit-transform: scale(1.15, 0.85);
                            transform: scale(1.15, 0.85);
                        }
                        input:checked ~ label {
                            background: #6f9eb1;
                        }
                        input:checked ~ label:after {
                            left: 20px;
                            background: #0085ba;
                        }
                        input:disabled ~ label {
                            background: #d5d5d5;
                            pointer-events: none;
                        }
                        input:disabled ~ label:after {
                            background: #bcbdbc;
                        }
                    
  </style>
</div>
`;

exports[`<ToggleSwitch /> should match snapshots 2`] = `
<div>
  <input
    aria-checked={true}
    checked={true}
    id="inline-toggle"
    onChange={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    type="checkbox"
  />
  <label
    data-tooltip="Test 2"
    htmlFor="inline-toggle"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    role="tooltip"
  />
  <style
    jsx={true}
  >
    
                        label {
                            position: relative;
                            display: block;
                            margin: 0 10px;
                            height: 20px;
                            width: 44px;
                            background: #898989;
                            border-radius: 100px;
                            cursor: pointer;
                            -webkit-transition: all 0.3s ease;
                            transition: all 0.3s ease;
                        }
                        label:after {
                            position: absolute;
                            left: -2px;
                            top: -3px;
                            display: block;
                            width: 26px;
                            height: 26px;
                            border-radius: 100px;
                            background: #fff;
                            -webkit-box-shadow: 0px 3px 3px rgba(0, 0, 0, 0.05);
                            box-shadow: 0px 3px 3px rgba(0, 0, 0, 0.05);
                            content: '';
                            -webkit-transition: all 0.3s ease;
                            transition: all 0.3s ease;
                        }
                        label:active:after {
                            -webkit-transform: scale(1.15, 0.85);
                            transform: scale(1.15, 0.85);
                        }
                        input:checked ~ label {
                            background: #6f9eb1;
                        }
                        input:checked ~ label:after {
                            left: 20px;
                            background: #0085ba;
                        }
                        input:disabled ~ label {
                            background: #d5d5d5;
                            pointer-events: none;
                        }
                        input:disabled ~ label:after {
                            background: #bcbdbc;
                        }
                    
  </style>
</div>
`;
